### ISSUE TITLE
Statistics page unefficient

### ISSUE THREAD

**Issue by lfortemps** [MISP/MISP#6243]
On an instance handling a lot of data, clicking on the Global Actions > statistics page will take a lot of time ( a couple of minutes ) and be executed in the main thread so the user needs to wait in front of a white screen. This is also very inefficient because each user clicking on the statistics is triggering the whole computation again. If multiple users are asking the statistics simultaneously, that can really have an impact on the server performances.

### Expected behavior

Click on the statistics page, have the result available in a reasonable time.

### Actual behavior

Click on the statistics page, wait for 3 minutes and finally get the page.

### Steps to reproduce the behavior

Have an instance with a large number of data, click on the statistics page. This will take a while and the server load will also significantly increase.

### Possible ideas to fix it

- Look for optimization in the SQL queries.
- Have the statistics cached for a given period of time (per organization)
- Compute the statistics using the workers and not the main thread
- Give the possibility to compute them beforehand (via crontab or similar).

**Comment by iglocska** [comment]
Correct, it was built with much smaller data-sets / user bases in mind. I am not a fan of purely relying on cron based pre-computation, for the simple reason that we have thousands of organisations running their instance, many of them running it for the most part unattended, so we'd end up not having the data-set available after an update at all for those communities.

Instead, decoupling the function, generating it when no valid / fresh cache exists whilst also making the function exposed to admins wanting to automate it via cron using a shell script sounds much more healthy.

So we can have a mixture of all 4 options:

- Obviously reviewing the SQL queries is a good idea, again these were built with tiny data sets in mind compared to what we have today
- cache the results / organisation for a set amount of time
- create a shell script for generating new caches (/app/Console/Command/EventShell.php is a good target)
- when a user views the statistics, check if the cache is outdated, if yes, trigger a new view, either in-line or via the background workers
- describe on the automation page how to run the shell script (for example `/var/www/MISP/app/Console cake Event generateStatistics [org_id]`)
